### PR TITLE
Ideas

### PR DESCRIPTION
**Pull request by meseta**
Integrates [Orteil's game ideas generator](https://orteil.dashnet.org/gamegen) (with permission), triggered using the `!idea` command.

### CODE REVIEW

File: src/commands/idea/orteil-gamegen.ts [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+  'to win the game', 'but doing so loses the game', 'or whatever']
+));
+
+// everything
+let allNames = [];
+let allActions = [];
+let allThings = [];
+let allGoals = [];
+
+for (let i in Genres) {
```
Comment (tonystr) [inline src/commands/idea/orteil-gamegen.ts]: Why looping `for (let i in Genres)`? Could use `for (const genre of Genres)`

File: src/commands/idea/orteil-gamegen.ts [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+  if (typeof Genres[i] === 'function') continue;
+  allNames = allNames.concat(Genres[i].name);
```
Comment (tonystr) [inline src/commands/idea/orteil-gamegen.ts]: Why concatinate arrays instead of `<Array>.push()`?

File: src/commands/idea/orteil-gamegen.ts [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+  allActions = allActions.concat(Genres[i].actions);
+  allThings = allThings.concat(Genres[i].things);
+  allGoals = allGoals.concat(Genres[i].goals);
+}
+
+export function generate(insane: boolean = false): string {
+  let genre = C(Genres);
+  let str = '';
```
Comment (tonystr) [inline src/commands/idea/orteil-gamegen.ts]: Can use ternary instead

File: src/commands/idea/orteil-gamegen.ts (hunk 1 above)
Comment (meseta) [reply src/commands/idea/orteil-gamegen.ts]: code is as close as possible to source code

File: src/commands/idea/orteil-gamegen.ts (hunk 2 above)
Comment (meseta) [reply src/commands/idea/orteil-gamegen.ts]: code is as close as possible to original source code

File: src/commands/idea/orteil-gamegen.ts (hunk 3 above)
Comment (meseta) [reply src/commands/idea/orteil-gamegen.ts]: ternary would reduce readability

**Pull request merged**